### PR TITLE
Add test coverage standard

### LINKED ISSUE #6: Code coverage standard
Explicitly stating what shouldn't be covered with a reason and the rest of the code should have 100% coverage

### PR DESCRIPTION
**Pull request by jdkandersson**
Closes #6

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -85,7 +85,29 @@ keeping in mind the option of breaking up the test into multiple tests.
 This structure makes it easy to understand what is required before test
 execution, how the test works and what it checks for in the end.
 
-# Non Compliant Code
+## Test Coverage
+
+Unit tests check whether the intended fucntionality has been implemented. This
```
Comment (arturo-seijas) [inline CONTRIBUTING.md:90]: You have a typo in here: fucntionality

**Review by arturo-seijas — CHANGES_REQUESTED** [review]
Just a minor thing

File: CONTRIBUTING.md (hunk 1 above)
Comment (jdkandersson) [reply CONTRIBUTING.md]: Thank you, fixed this one and another one

**Review by arturo-seijas — APPROVED** [review]
LGTM

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -85,7 +85,29 @@ keeping in mind the option of breaking up the test into multiple tests.
...
+main, such as through a pull request. Any coverage exclusion should have an
+explanatory comment, such as:
+
+```Python
+# Exclude from coverage since unit tests should not run as __main__
+if __name__ == "__main__":  # pragma: no cover
+   ...
+```
+
+This ensures a high coverage minimum and no coverage regression.
```
Comment (mthaddon) [inline CONTRIBUTING.md:108]: Should we talk about the mechanism that we plan to use for this, i.e. in Indico currently we have this in the `tox.ini`:

```
[testenv:unit]
description = Run unit tests
deps =
    pytest
    coverage[toml]
    -r{toxinidir}/requirements.txt
commands =
    coverage run --source={[vars]src_path} \
        -m pytest --ignore={[vars]tst_path}integration -v --tb native -s {posargs}
    coverage report --fail-under=90
```

Comment (jdkandersson) [reply CONTRIBUTING.md:108]: Yes, there is a setting in `[tool.coverage.report]` for `pyproject.toml` called `fail_under`, I'll propose we use that to keep configuration in one place

**Review by mthaddon — APPROVED** [review]
LGTM, thx!

**Pull request merged**